src/main/java/core/basesyntax/service/impl/PurchaseOperationHandler.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.Operation;
+import core.basesyntax.service.OperationHandler;
+import core.basesyntax.service.ShopDao;
+import java.util.Map;
+
+public class PurchaseOperationHandler extends OperationHandler {
+    public PurchaseOperationHandler(ShopDao dao) {
+        super(dao);
+    }
+
+    @Override
+    public void process(Operation operation) {
+        Map.Entry<String, Integer> shopEntry = dao.get(operation.getFruitName());
+        if (shopEntry.getValue() < operation.getAmount()) {
```

Comment:
I see one problem.
What will happen if you have 20 bananas and someone wants to buy 20 bananas?